services/index/src/api.js

```diff
@@ -57,35 +54,31 @@ builder.declare({
   let namespace = indexPath.join('.');
 
   // Load indexed task
-  let tasks;
   try {
-    tasks = await this.IndexedTask.query({
-      namespace: namespace,
-      name: name,
-      expires: Entity.op.greaterThan(new Date()),
-    });
```

Comment:
I think this only used `query` here instead of `load` to validate that the row had not expired.  There can only ever be one row.  So, I think this could be translated to a call to `get_indexed_task`.  No pagination required..

Also, the updated code should check the expiration in JS.

Reply:
> So, I think this could be translated to a call to get_indexed_task. No pagination required..

Sounds good, makes sense.

> Also, the updated code should check the expiration in JS.

Why should it check the expiration in JS? All of the invocations of load and query for the indexed task have `expires: Entity.op.greaterThan(new Date()),` in them. Having the check in the JS will require some refactoring which may introduce bugs. Thoughts?

Reply:
It's a pretty minor refactor -- just adding an `if` statement, right?  And this is well-covered by tests, at least the normal (non-expired) case.  Getting an expired row is definitely the unusual case here.

To be honest, even with Azure I'm not sure why we used `.query` instead of `.load` with a check of the expires value.